docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,882 @@
+node BGP speakers do not accept routes to NodeSubnets by default. However, to
+route east-west traffic in no-overlay mode, these routes must be imported to the
+host routing table. We need to update the node FRRConfiguration generation logic
+to check whether a network is in no-overlay mode and generate the node
+FRRConfiguration accordingly.
+
+#### No SNAT for egress traffic
+
+The OVN-Kubernetes BGP feature currently disables SNAT only for traffic destined
+to external networks, not for inter-pod communication across nodes.
+
+In no-overlay mode, the goal is to fully leverage BGP and eliminate the SNAT
+step on the gateway router, even for intra-cluster traffic. Within the same
+network, pod-to-pod traffic will utilize BGP-learned routes to reach its
+destination node directly, bypassing Geneve encapsulation and SNAT to the
+nodeIP.
+
+The only exception is for traffic destined to other nodes in the cluster
+(pod-to-other_node), to ensure nodePort services can be accessed across
+networks. This egress traffic will remain SNATed to the node IP before leaving
+the node. This is the current behavior when a network is advertised. We will not
+change it for no-overlay mode.
+
+#### Active traffic blocking between UDNs
+
+The [existing UDN isolation
+mechanism](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5186/commits/8f6e7d30ee5f4926a21e2de75488aad80344814b)
+will still be in place. Pods in different UDNs will be isolated from each other
+and the default network, utilizing the existing OVN ACLs on every node logical
+switch: ACLs are enforced on outgoing traffic at each advertised UDN switch,
+verifying whether both source and destination IPs belong to the same advertised
+UDN. If the destination IP does not come from the same UDN subnet as the source
+IP, traffic is dropped. No-overlay mode will not affect the isolation between
+UDNs or between UDNs and the default network. So, even if multiple UDNs were
+advertised to the same VRF, the cross-UDN traffic is still blocked. This is the
+current behavior when a network is advertised. We will not change it for no
+overlay mode.
+
+### Workflow
+
+#### Enable No-overlay Mode for the Default Network
+
+1. The frr-k8s pod shall be deployed on each node.  
+2. Before starting the ovnkube pods, the cluster admin shall create a base
+   FRRConfiguration CR that is used for generating the per-node FRRConfiguration
+   by OVN-K.
+
+    ```yaml
+    apiVersion: frrk8s.metallb.io/v1beta1
+    kind: FRRConfiguration
+    metadata:
+      name: default
+      namespace: frr-k8s-system
+      label:
+        # A custom label
+        network: default
+    spec:
+      bgp:
+        routers:
+        - asn: 64512
+          neighbors:
+          # the external BGP route reflector
+          - address: 172.20.0.2
+            asn: 64512
+            disableMP: true
+    ```
+
+3. The cluster admin shall create a RouteAdvertisements CR to advertise the
```

Comment:
hmm am I confusing something? I don't think creating RA CR is a requirement for no-overlay mode right? I think perhaps you mean to update the enhancement around this ? like we discussed in BGP sync meeting I think we want to create things internally for the user rather than expect them to have RA created etc.. which has a totally different API meaning.
I assumed RA today means:

1. advertise pod networks to outside the cluster elements

and this meaning of the API mustn't change IMHO

1. advertise pod networks to other nodes in the cluster -> this is the semantic of the new field you are adding which should require no additional end-user API changes